src/features/transactions-graph/components/transactions-graph.tsx

```diff
@@ -15,63 +19,89 @@ export function TransactionsGraph({ transactionsGraphData }: Props) {
+          }}
+        >
+          <div>{/* The first header cell is empty */}</div>
+          {verticals.map((vertical, index) => (
+            <div className={cn('p-2 flex justify-center')} key={index}>
+              <VerticalTitle vertical={vertical} />
+            </div>
+          ))}
+        </div>
+        <div
+          className={cn('relative grid')}
+          style={{
+            gridTemplateColumns: `${gridTemplateColumns}`,
+            gridTemplateRows: `repeat(${horizontalsCount}, ${graphConfig.rowHeight}px)`,
+          }}
+        >
+          {/* The below div is for drawing the background dash lines */}
+          <div className={cn('absolute left-0')}>
+            <div>
+              <div className={cn('p-0')}></div>
               <div
-                className={cn('grid h-full')}
+                className={cn('p-0')}
                 style={{
-                  gridTemplateColumns: `${gridTemplateColumns}`,
                   height: `${horizontalsCount * graphConfig.rowHeight}px`,
+                  width: `${graphConfig.colWidth * verticalsCount}px`,
                 }}
               >
-                <div></div>
-                {verticals
-                  .filter((a) => a.type !== 'Placeholder') // Don't need to draw for the empty vertical
-                  .map((_, index) => (
-                    <div key={index} className={cn('flex justify-center')}>
-                      <div className={cn('border h-full border-dashed')} style={{ borderLeftWidth: graphConfig.lineWidth }}></div>
-                    </div>
-                  ))}
+                <div
+                  className={cn('grid h-full')}
+                  style={{
+                    gridTemplateColumns: `${gridTemplateColumns}`,
+                    height: `${horizontalsCount * graphConfig.rowHeight}px`,
+                  }}
+                >
+                  <div></div>
+                  {verticals
+                    .filter((a) => a.type !== 'Placeholder') // Don't need to draw for the empty vertical
+                    .map((_, index) => (
+                      <div key={index} className={cn('flex justify-center')}>
+                        <div className={cn('border h-full border-dashed')} style={{ borderLeftWidth: graphConfig.lineWidth }}></div>
+                      </div>
+                    ))}
+                </div>
               </div>
             </div>
           </div>
+          {horizontals.map((row, index) => (
+            <Horizontal key={index} verticals={verticals} horizontal={row} />
+          ))}
         </div>
-        {horizontals.map((row, index) => (
-          <Horizontal key={index} verticals={verticals} horizontal={row} />
-        ))}
       </div>
-    </div>
+      <div className="sticky bottom-0 left-full z-50 flex size-0 overflow-visible">
+        <Button id="download-transactions-visual" className="absolute bottom-1 right-0 w-32" variant="outline" onClick={downloadImage}>
```

Comment:
out of curiosity, do you need `id="download-transactions-visual"` here? It doesn't seem to be referenced anywhere else.

Reply:
Added a comment for clarity.